extra/html-extra/HtmlExtension.php

```diff
@@ -108,4 +112,75 @@ function twig_html_classes(...$args): string
 
     return implode(' ', array_unique($classes));
 }
+
+function twig_attr_merge(...$arrays): array
+{
+    $result = [];
+
+    foreach ($arrays as $argNumber => $array) {
+        if (!$array) {
+            continue;
+        }
+
+        if (!twig_test_iterable($array)) {
+            throw new RuntimeError(sprintf('The attr_merge filter only works with arrays or "Traversable", got "%s" for argument %d.', \gettype($array), $argNumber + 1));
+        }
+
+        $array = twig_to_array($array);
+
+        foreach (['class', 'style', 'data'] as $deepMergeKey) {
+            if (isset($array[$deepMergeKey])) {
+                $value = $array[$deepMergeKey];
+                unset($array[$deepMergeKey]);
+
+                if (!twig_test_iterable($value)) {
+                    $value = (array) $value;
+                }
+
+                $value = twig_to_array($value);
+
+                $result[$deepMergeKey] = array_merge($result[$deepMergeKey] ?? [], $value);
+            }
+        }
+
+        $result = array_merge($result, $array);
+    }
+
+    return $result;
+}
+
+function twig_attr(Environment $env, ...$args): string
+{
+    $attr = twig_attr_merge(...$args);
+
+    if (isset($attr['class'])) {
+        $attr['class'] = trim(implode(' ', array_values($attr['class'])));
+    }
+
+    if (isset($attr['style'])) {
+        $style = '';
+        foreach ($attr['style'] as $name => $value) {
+            if (is_numeric($name)) {
+                $style .= $value.'; ';
+            } else {
+                $style .= $name.': '.$value.'; ';
+            }
+        }
+        $attr['style'] = trim($style);
+    }
+
+    if (isset($attr['data'])) {
+        foreach ($attr['data'] as $name => $value) {
+            $attr['data-'.$name] = $value;
+        }
+        unset($attr['data']);
+    }
+
+    $result = '';
+    foreach ($attr as $name => $value) {
+        $result .= twig_escape_filter($env, $name, 'html_attr').'="'.htmlspecialchars($value).'" ';
```

Comment:
Let's consider `false` as a way to disable the attribute? And `true` would not generate the value part?

Reply:
I guess that makes sense. I'd have `false` omit the attribute output altogether. For `true`, I'd use something like `name="name"` for backwards compat with (X)HTML.

Reply:
⚠️  "true" must be written "true" for some enumerated aria attributes. 

Almost everywhere else, indeed, `false` should remove/hide the attribute.

`""` should not happen, i guess, but should be dealt with attention, because the following have all the same meaning: 

* `disabled=""` 
* `disabled` 
* `disabled="false"`
* `disabled="disabled"`

Reply:
For the aria cases, _**if**_ you want,  there is some classic cases + docs here

https://github.com/symfony/ux/blob/647532ab688f79acfbcb1c895e88a8b2f1a502f6/src/Icons/src/Icon.php#L139-L162

with some test cases if you need too https://github.com/symfony/ux/blob/647532ab688f79acfbcb1c895e88a8b2f1a502f6/src/Icons/tests/Unit/IconTest.php#L226 


(other similar in the TwigComponent / ComponentAttributes)

Reply:
Except for `aria-*` attributes that support keyword and enumerated values. In some cases `aria-*` attributes support `true` and `false` string values.

* https://hidde.blog/boolean-attributes-in-html-and-aria-whats-the-difference/
* https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#keywords-and-enumerated-attributes
* https://www.w3.org/TR/wai-aria-1.1/#typemapping

Reply:
Fortunately, Twig and PHP have separate types for strings and booleans. Passing the string `true` or `false` will not trigger the special logic for boolean values.

Reply:
@stof I guess what I was trying to say is that yes you could pass `'true'` or '`false`' as parameters but this would require string coercion when setting the value.

 The previous comment stated that:

> Almost everywhere else, indeed, false should remove/hide the attribute.

I was pointing out that `aria-*` attributes do accept a string value of `'false'` and in those cases a coercion of bool `false` to string `'false'` might be warranted. This could also apply to `data-*` attributes.

IMO:

* `null` value always removes the attribute from the final string rendering.
* `true` and `false` values for `aria-*` attributes should coerce boolean  to their string values `'true'` and `'false'`
* `true` and `false` values for all other attributes should act as boolean html attributes. ie true renders the attribute, false removes the attribute (as per @fabpot [comment](https://github.com/twigphp/Twig/pull/3930#discussion_r1777667412)).